caikit_nlp/modules/text_embedding/embedding.py

```diff
@@ -0,0 +1,142 @@
+
+logger = alog.use_channel("TXT_EMB")
+error = error_handler.get(logger)
+
+
+@module(
+    "EEB12558-B4FA-4F34-A9FD-3F5890E9CD3F",
+    "EmbeddingModule",
+    "0.0.1",
+    EmbeddingTask,
+)
+class EmbeddingModule(ModuleBase):
+
+    _ARTIFACTS_PATH_KEY = "artifacts_path"
+    _ARTIFACTS_PATH_DEFAULT = "artifacts"
+
+    def __init__(
+        self,
+        model: SentenceTransformer,
+    ):
+        super().__init__()
+        self.model = model
+
+    @classmethod
+    def load(cls, model_path: str, *args, **kwargs) -> "EmbeddingModule":
+        """Load model
+
+        Args:
+            model_path: str
+                Path to the config dir under the model_id (where the config.yml lives)
+
+        Returns:
+            EmbeddingModule
+                Instance of this class built from the model.
+        """
+
+        config = ModuleConfig.load(model_path)
+        artifacts_path = config.get(cls._ARTIFACTS_PATH_KEY)
+
+        error.value_check(
+            "<NLP07391618E>",
+            artifacts_path,
+            ValueError(f"Model config missing '{cls._ARTIFACTS_PATH_KEY}'"),
+        )
+
+        artifacts_path = os.path.abspath(os.path.join(model_path, artifacts_path))
+        error.dir_check("<NLP34197772E>", artifacts_path)
+
+        return cls.bootstrap(model_name_or_path=artifacts_path)
+
+    def run(
+        self, input: str, **kwargs  # pylint: disable=redefined-builtin
+    ) -> EmbeddingResult:
+        """Run inference on model.
+        Args:
+            input: str
+                Input text to be processed
+        Returns:
+            EmbeddingResult: the result vector nicely wrapped up
+        """
+        error.type_check("<NLP27491611E>", str, input=input)
+
+        return EmbeddingResult(Vector1D.from_vector(self.model.encode(input)))
+
+    @classmethod
+    def bootstrap(cls, model_name_or_path: str) -> "EmbeddingModule":
+        """Bootstrap a sentence-transformers model
+
+        Args:
+            model_name_or_path: str
+                Model name (Hugging Face hub) or path to model to load.
+        """
+        return cls(model=SentenceTransformer(model_name_or_path=model_name_or_path))
+
+    def save(self, model_path: str, *args, **kwargs):
+        """Save model using config in model_path
+
+        Args:
+            model_path: str
+                Path to model config
+        """
+
+        model_config_path = model_path  # because the param name is misleading
+
+        error.type_check("<NLP82314992E>", str, model_path=model_config_path)
+        error.value_check(
+            "<NLP40145207E>",
+            model_config_path is not None and model_config_path.strip(),
+            f"model_path '{model_config_path}' is invalid",
+        )
+
+        model_config_path = os.path.abspath(
+            model_config_path.strip()
+        )  # No leading/trailing spaces sneaky weirdness
+
+        os.makedirs(model_config_path, exist_ok=False)
```

Comment:
We can also use `saver.add_dirs` functions from here: https://github.com/caikit/caikit/blob/a16f063f6155f0088eb9959a32b7f0871e89731d/caikit/core/modules/saver.py#L117

This will avoid need to also make the absolute path etc above.

Reply:
We can also use the context base use of `ModuleSaver` and that can take care of adding / making the directory as well.  Example: https://github.com/caikit/caikit-nlp/blob/main/caikit_nlp/modules/text_generation/peft_prompt_tuning.py#L472

Reply:
> We can also use `saver.add_dirs` functions from here: https://github.com/caikit/caikit/blob/a16f063f6155f0088eb9959a32b7f0871e89731d/caikit/core/modules/saver.py#L117
> 
> This will avoid need to also make the absolute path etc above.

os.makedirs() is used because in-place updates are not safe to run on existing dirs.  I use exist_ok=False to enforce that where ModuleSaver would not and can remove existing file trees on exceptions.  Not cool.

The abspath() is per ModuleSaver param docstrings.  strip() should not be necessary but seems better than testing what blanks would do for user experience.

Reply:
> We can also use the context base use of `ModuleSaver` and that can take care of adding / making the directory as well. Example: https://github.com/caikit/caikit-nlp/blob/main/caikit_nlp/modules/text_generation/peft_prompt_tuning.py#L472

context for ModuleSaver is pretty dangerous here.  Use save() wrong and ModuleSaver will wipe out a directory that might be important.

Reply:
but that is related to `os.mkdirs` only right? So that can be fixed by adding this an option in the context manager or in the caikit repo itself? May be club it with your DM PR in caikit ?